cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -94,7 +94,7 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 	// Construct a descriptor for the current state of the world
 	beforeState := ClusterState{
 		KubeVersion:     clusterVersionStr,
-		DNSVersion:      dns.GetKubeDNSVersion(clusterVersion),
+		DNSVersion:      dns.GetDNSVersion(clusterVersion, "kube-dns"),
```

Comment:
Check whether the CoreDNS feature gate is enabled and optionally show the `coredns` label and version here please